cdap-data-fabric/src/main/java/co/cask/cdap/data2/sql/PostgresSqlStructuredTableAdmin.java

```diff
@@ -0,0 +1,127 @@
+/*
+ * Copyright © 2019 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.data2.sql;
+
+import co.cask.cdap.common.AlreadyExistsException;
+import co.cask.cdap.spi.data.StructuredTableAdmin;
+import co.cask.cdap.spi.data.table.StructuredTableId;
+import co.cask.cdap.spi.data.table.StructuredTableSpecification;
+import co.cask.cdap.spi.data.table.StructuredTableSpecificationRegistry;
+import co.cask.cdap.spi.data.table.field.FieldType;
+import com.google.common.base.Joiner;
+
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.DatabaseMetaData;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import javax.annotation.Nullable;
+
+/**
+ * Sql structured admin to use jdbc connection to create and drop tables.
+ */
+public class PostgresSqlStructuredTableAdmin implements StructuredTableAdmin {
+  private final Connection connection;
+
+  public PostgresSqlStructuredTableAdmin(Connection connection) {
+    this.connection = connection;
+  }
+
+  @Override
+  public void create(StructuredTableSpecification spec) throws IOException, AlreadyExistsException {
+    try (Statement statement = connection.createStatement()) {
+      DatabaseMetaData metaData = connection.getMetaData();
+      // Postgres is case insensitive be default
+      ResultSet rs = metaData.getTables(null, null,
```

Comment:
should we make our table-names case-insensitive then? Or require that they are lower case?

Reply:
I will make the table name case-insensitive, don't see a big reason to require they are all lower cased.